### PR TITLE
[Merged by Bors] - feat(MvPolynomial): more lemmas about `finSuccEquiv`

### PR DESCRIPTION
**Pull request by YaelDillies**
Co-authored-by: Andrew Yang

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/MvPolynomial/Equiv.lean [hunk 1]
```diff
@@ -441,7 +440,7 @@ lemma totalDegree_coeff_finSuccEquiv_add_le (f : MvPolynomial (Fin (n + 1)) R) (
   · rw [← support_coeff_finSuccEquiv]
     exact hσ1
 
-theorem finSuccEquiv_support (f : MvPolynomial (Fin (n + 1)) R) :
+theorem support_finSuccEquiv {f : MvPolynomial (Fin (n + 1)) R} :
```
Comment (b-mehta) [inline Mathlib/Algebra/MvPolynomial/Equiv.lean:443]: Let's keep the implicitness unchanged here: as a rewrite lemma it's useful to be able specify `f`.

File: Mathlib/Algebra/MvPolynomial/Equiv.lean [hunk 2]
```diff
@@ -454,8 +453,14 @@ theorem finSuccEquiv_support (f : MvPolynomial (Fin (n + 1)) R) :
...
-theorem finSuccEquiv_support' {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} :
-    Finset.image (Finsupp.cons i) (Polynomial.coeff ((finSuccEquiv R n) f) i).support =
+@[deprecated (since := "2024-11-05")] alias finSuccEquiv_support := support_finSuccEquiv
+
+theorem mem_support_finSuccEquiv {f : MvPolynomial (Fin (n + 1)) R} {x} :
+    x ∈ (finSuccEquiv R n f).support ↔ x ∈ (fun m : Fin (n + 1) →₀ _ ↦ m 0) '' f.support := by
+  simpa using congr(x ∈ $support_finSuccEquiv)
+
+theorem image_support_finSuccEquiv {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} :
+    Finset.image (Finsupp.cons i) (Polynomial.coeff (finSuccEquiv R n f) i).support =
```
Comment (b-mehta) [inline Mathlib/Algebra/MvPolynomial/Equiv.lean:463]:
```suggestion
    (Polynomial.coeff (finSuccEquiv R n f) i).support.image (Finsupp.cons i) =
```
If this works (untested) it looks easier to read, and makes the name change look more reasonable

Comment (b-mehta) [reply Mathlib/Algebra/MvPolynomial/Equiv.lean:463]: (plus dot notation for Polynomial.coeff, if that works too)

**Review by b-mehta — APPROVED** [review]

File: Mathlib/Algebra/MvPolynomial/Equiv.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/MvPolynomial/Equiv.lean:443]: Not sure why I did this change. Looks like copy-pasta from `mem_support_finSuccEquiv`

File: Mathlib/Algebra/MvPolynomial/Equiv.lean [hunk 3]
```diff
@@ -414,7 +413,7 @@ theorem coeff_eval_eq_eval_coeff (s' : Fin n → R) (f : Polynomial (MvPolynomia
   simp only [Polynomial.coeff_map]
 
 theorem support_coeff_finSuccEquiv {f : MvPolynomial (Fin (n + 1)) R} {i : ℕ} {m : Fin n →₀ ℕ} :
-    m ∈ (Polynomial.coeff ((finSuccEquiv R n) f) i).support ↔ Finsupp.cons i m ∈ f.support := by
+    m ∈ (Polynomial.coeff (finSuccEquiv R n f) i).support ↔ Finsupp.cons i m ∈ f.support := by
```
Comment (b-mehta) [inline Mathlib/Algebra/MvPolynomial/Equiv.lean:416]:
```suggestion
    m ∈ ((finSuccEquiv R n f).coeff i).support ↔ Finsupp.cons i m ∈ f.support := by
```
untested. Maybe `m.cons i` too on the rhs?

**Review by b-mehta — COMMENTED** [review]
bors d+

File: Mathlib/Algebra/MvPolynomial/Equiv.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/Algebra/MvPolynomial/Equiv.lean]: Whoops sorry I thought I'd gotten them all

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**